roles/network_plugin/weave/defaults/main.yml

```diff
@@ -4,3 +4,6 @@ weave_memory_limit: 400M
 weave_cpu_limit: 30m
 weave_memory_requests: 64M
 weave_cpu_requests: 10m
+
+# this variable is use in seed mode
+weave_ip_current_cluster: '{% for host in groups["k8s-cluster"] %}{{ hostvars[host]["ansible_default_ipv4"]["address"] }}{% if not loop.last %} {% endif %}{% endfor %}'
```

Comment:
You cannot use hostvars before they have been gathered. defaults task is evaluated before setup. This task should be done in `roles/network_plugin/weave/tasks/seed.yml`.

Secondly, host ips are set by the `ip` var, with fallback to ansible default ipv4 ip. This line should read:
`weave_ip_current_cluster: '{% for host in groups["k8s-cluster"] %}{{ hostvars[host]['ip'] | default(hostvars[host]['ansible_default_ipv4']['address']) }}{% if not loop.last %} {% endif %}{% endfor %}'`